src/WebJobs.Script/OutOfProc/Rpc/LanguageWorkerChannel.cs

```diff
@@ -476,5 +476,13 @@ private bool IsTriggerMetadataPopulatedByWorker()
         {
             return !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(LanguageWorkerConstants.RpcHttpTriggerMetadataRemoved));
         }
+
+        public async Task DrainInvocationsAsync()
+        {
+            foreach (ScriptInvocationContext currContext in _executingInvocations.Values)
+            {
+                await currContext.ResultSource.Task;
```

Comment:
Why wouldn't we want to enforce timeout on the ShutdownManager?

Reply:
Enforcing timeout where we are actually waiting on a task is  better. Purpose of FunctionDispatcherShutdownManager is to hook into DI pipeline so that we can call StopAsync before we hit dispose